src/windows.rs

```diff
@@ -0,0 +1,464 @@
+use crate::Error;
+use async_trait::async_trait;
+use futures::Stream;
+use maplit::hashmap;
+use std::collections::HashMap;
+use std::ffi::{c_void, OsString};
+use std::iter;
+use std::os::windows::prelude::*;
+
+use brightness_windows::Windows::Win32::{
+    Devices::Display::{
+        DestroyPhysicalMonitor, GetMonitorBrightness, GetNumberOfPhysicalMonitorsFromHMONITOR,
+        GetPhysicalMonitorsFromHMONITOR, SetMonitorBrightness, DISPLAYPOLICY_AC, DISPLAYPOLICY_DC,
+        DISPLAY_BRIGHTNESS, IOCTL_VIDEO_QUERY_DISPLAY_BRIGHTNESS,
+        IOCTL_VIDEO_QUERY_SUPPORTED_BRIGHTNESS, IOCTL_VIDEO_SET_DISPLAY_BRIGHTNESS,
+    },
+    Foundation::{CloseHandle, BOOL, HANDLE, LPARAM, PWSTR, RECT},
+    Graphics::Gdi::{
+        EnumDisplayDevicesW, EnumDisplayMonitors, GetMonitorInfoW, DISPLAY_DEVICEW, HDC, HMONITOR,
+        MONITORINFO, MONITORINFOEXW,
+    },
+    Storage::FileSystem::{
+        CreateFileW, FILE_ACCESS_FLAGS, FILE_FLAGS_AND_ATTRIBUTES, FILE_SHARE_MODE, OPEN_EXISTING,
+    },
+    System::Diagnostics::Debug::{GetLastError, ERROR_ACCESS_DENIED, ERROR_NOT_FOUND, WIN32_ERROR},
+    System::SystemServices::{DeviceIoControl, GENERIC_READ, GENERIC_WRITE},
+    UI::WindowsAndMessaging::EDD_GET_DEVICE_INTERFACE_NAME,
+};
+
+#[allow(non_snake_case)]
+#[derive(Debug)]
+pub struct Brightness {
+    hPhysicalMonitor: HANDLE,
+    file_handle: HANDLE,
+    device_name: String,
+    device_description: String,
+    device_id: String,
+    device_key: String,
+    device_string: String,
+}
+
+impl Drop for Brightness {
+    fn drop(&mut self) {
+        unsafe {
+            DestroyPhysicalMonitor(self.hPhysicalMonitor);
+            CloseHandle(self.file_handle);
+        }
+    }
+}
+
+#[async_trait]
+impl crate::Brightness for Brightness {
+    async fn device_name(&self) -> Result<String, Error> {
+        Ok(self.device_name.clone())
+    }
+
+    async fn device_info(&self) -> Result<HashMap<String, String>, Error> {
+        Ok(hashmap! {
+            "device_description".to_string() => self.device_description.clone(),
+            "device_id".to_string() => self.device_id.clone(),
+            "device_key".to_string() => self.device_key.clone(),
+            "device_string".to_string() => self.device_string.clone(),
+        })
+    }
+
+    async fn get(&self) -> Result<u32, Error> {
+        let ioctl_query = ioctl_query_supported_brightness(self);
+        return match ioctl_query {
+            Ok(_) => ioctl_query_display_brightness(self).map_err(|e| e.into()),
+            Err(e) if e == ERROR_NOT_FOUND || e == WIN32_ERROR(6) => {
```

Comment:
Is `6` `ERROR_INVALID_HANDLE`? If yes, can the named constant be used?

Reply:
Also, does the following compile? (No access to Windows right now)
```rust
Err(ERROR_NOT_FOUND | ERROR_INVALID_HANDLE) => ...
```

Reply:
yep, that works

Reply:
Thank you.